signing-jobs/build/sign-artifacts/Jenkinsfile

```diff
@@ -0,0 +1,127 @@
+                        defaultValue: ""
+                    ],
+                    [
+                        name: 'DRY_RUN',
+                        description: 'Only do dry run test and exit.',
+                        $class: 'BooleanParameterDefinition',
+                        defaultValue: false
+                    ],
+                    [
+                        name: 'ENV',
+                        description: 'Which environment to sign in',
+                        $class: 'hudson.model.ChoiceParameterDefinition',
+                        choices: [
+                            "stage",
+                            "prod",
+                        ].join("\n"),
+                        defaultValue: "stage"
+                    ],
+                    [
+                        name: 'KEY_NAME',
+                        description: 'Which key to sign with (if ENV==stage, everything becomes "test")',
+                        $class: 'hudson.model.ChoiceParameterDefinition',
+                        choices: [
+                            "test",
+                            "beta2",
+			    "redhatrelease2",
+                        ].join("\n"),
+                        defaultValue: "test"
+                    ],
+                    commonlib.mockParam(),
+                ]
+            ],
+            disableConcurrentBuilds()
+        ]
+    )
+
+    commonlib.checkMock()
+    buildlib.cleanWorkdir(env.WORKSPACE)
+
+    // must be able to access remote registry for verification
+    buildlib.registry_quay_dev_login()
+
+    stage('sign-artifacts') {
+	def noop = params.DRY_RUN ? " --noop" : " "
+
+	wrap([$class: 'BuildUser']) {
+	    echo "Submitting signing requests as user: ${env.BUILD_USER}"
+
+	    withCredentials([file(credentialsId: 'msg-openshift-art-signatory-prod.crt', variable: 'busCertificate'),
+			     file(credentialsId: 'msg-openshift-art-signatory-prod.key', variable: 'busKey')]) {
+		echo "Authenticating with bus cert/key at: ${busCertificate}/${busKey}"
+		// Following line is for debugging when we're
+		// initially testing this
+		sh 'ls -l ${busCertificate} ${busKey}'
+
+		// ######################################################################
+		def baseUmbParams = buildlib.cleanWhitespace("""
+                    --requestor ${env.BUILD_USER} --sig-keyname ${params.KEY_NAME}
+                    --release-name ${params.NAME} --client-cert ${busCertificate}
+                    --client-key ${busKey} --env ${params.ENV}
+                    ${noop}
+                """)
+
+		// ######################################################################
+		def openshiftJsonSignParams = buildlib.cleanWhitespace("""
+		    ${baseUmbParams}
+		    --product openshift
+		    --request-id 'openshift-json-digest ${env.BUILD_URL}'
+                """)
+
+		echo "Submitting OpenShift Payload JSON claim signature request"
+		res = commonlib.shell(
+		    returnAll: true,
```

Comment:
This will always succeed. Shouldn't we check the `returnStatus`?

Reply:
That's a good point!